pkg/ui/handlers.go

```diff
@@ -1,37 +1,80 @@
 package ui
 
 import (
+	"fmt"
 	"io/fs"
 	"net/http"
-	"path"
+	"regexp"
+	"strings"
 
 	"github.com/canonical/identity-platform-login-ui/internal/logging"
 	"github.com/go-chi/chi/v5"
 )
 
 type API struct {
-	fileServer http.Handler
-
-	logger logging.LoggerInterface
+	fileServer   http.Handler
+	fileSystem   fs.FS
+	servePattern *regexp.Regexp
+	htmlRoutes   []string
+	logger       logging.LoggerInterface
 }
 
 func (a *API) RegisterEndpoints(mux *chi.Mux) {
 	// TODO @shipperizer unsure if we deal with any POST/PUT/PATCH via js
-	mux.HandleFunc("/*", a.uiFiles)
+	mux.HandleFunc("/ui/*", a.uiFiles)
+
+}
+
+func (a *API) getHtmlRoutes() []string {
+	routes := make([]string, 0)
+	nodeList, _ := fs.ReadDir(a.fileSystem, ".")
+
+	for _, node := range nodeList {
+		if match := a.servePattern.FindString(node.Name()); match == ".html" {
+			route := node.Name()[:len(node.Name())-len(match)]
+			routes = append(routes, fmt.Sprintf("/ui/%s", route))
+			continue
+		}
+	}
+	return routes
+}
```

Comment:
IMO this is a little complicated and adds little value to the code base. 

If the only reason we do this is to remove the `.html` from the prometheus labels I think that it makes most sense to address it in the Prometheus middleware.

Reply:
ok, I removed it.